node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -51,17 +52,32 @@ spec = giveGtConf $ giveNodeConf $ giveInfraConf $ giveUpdateConf $ giveCoreConf
     describe "Client.Txp.Util" $ do
         describe "createMTx" $ createMTxSpec
 
+-- GHC doesn't support impredicative polymorphism so we need a wrapper
+-- for the list below to typecheck.
+data TestFunctionWrapper
+    = forall prop. (Testable prop) => TestFunctionWrapper (InputSelectionPolicy -> prop)
```

Comment:
This is neat, but you put only those properties to that list which can depend on `InputSelectionPolicy`, so essentially anything except `TxpTestProperty` hardly can be used there

Reply:
I wouldn't go for it, but there was one function that required it:

    feeForManyAddressesSpec :: InputSelectionPolicy -> Bool -> TxpTestProperty ()

Reply:
Oh harsh, okay